src/compiler/checker.ts

```diff
@@ -37573,15 +37575,15 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
         return checkSatisfiesExpressionWorker(node.expression, node.type);
     }
 
-    function checkSatisfiesExpressionWorker(expression: Expression, target: TypeNode, checkMode?: CheckMode) {
+    function checkSatisfiesExpressionWorker(expression: Expression, target: TypeNode, checkMode = CheckMode.Normal) {
         const exprType = checkExpression(expression, checkMode);
         const targetType = getTypeFromTypeNode(target);
         if (isErrorType(targetType)) {
             return targetType;
         }
         const errorNode = findAncestor(target.parent, n => n.kind === SyntaxKind.SatisfiesExpression || n.kind === SyntaxKind.JSDocSatisfiesTag);
         checkTypeAssignableToAndOptionallyElaborate(exprType, targetType, errorNode, expression, Diagnostics.Type_0_does_not_satisfy_the_expected_type_1);
-        return exprType;
+        return checkExpression(expression, checkMode | CheckMode.SatisfiesOutput);

```

Comment:
`exprType` is used to check the assignability, it may contain "resolved" types for narrowable references. So the assignability works like it works in:
```ts
const x: "A" = genericReferenceNarrowedDownToA;
```

But the output type is different because we should propagate that "resolved" type in:
```ts
// actual on main: `"A"`
// expected: `AB extends "A" | "B"`
const x = genericReferenceNarrowedDownToA satisfies "A";
```